### PR TITLE
removing persistent workers, erroring if discovered

### PR DESCRIPTION
**Pull request by goddenrich**
This removes the ability to use persistent workers. Workers are no longer started or stopped. The worker commands are still parsed but if they are encountered the build or test will error.

### CODE REVIEW

File: src/build/build_step.go [hunk 1]
```diff
@@ -1215,57 +1211,16 @@ func (r *progressReader) Read(b []byte) (int, error) {
...
 
-// buildMaybeRemotely builds a target, either sending it to a remote worker if needed,
-// or locally if not.
-func buildMaybeRemotely(state *core.BuildState, target *core.BuildTarget, inputHash []byte) (*core.BuildMetadata, error) {
+// build builds a target locally, it errors if a remote worker is needed since this has beeen removed.
+func build(state *core.BuildState, target *core.BuildTarget, inputHash []byte) (*core.BuildMetadata, error) {
 	metadata := new(core.BuildMetadata)
 
-	workerCmd, workerArgs, localCmd, err := core.WorkerCommandAndArgs(state, target)
+	workerCmd, _, localCmd, err := core.WorkerCommandAndArgs(state, target)
```
Comment (samwestmoreland) [inline src/build/build_step.go:1218]: Should we delete the code in `WorkerCommandAndArgs` as well? Looks like it's to do with `$(worker)` which I guess we don't need anymore

File: src/test/test_step.go [hunk 2]
```diff
@@ -546,24 +545,15 @@ func moveOutputFile(state *core.BuildState, hash []byte, from, to, dummy string)
 	return fs.RecordAttr(to, hash, xattrName, state.XattrsSupported)
 }
 
-// startTestWorkerIfNeeded starts a worker server if the test needs one.
-func startTestWorkerIfNeeded(tid int, state *core.BuildState, target *core.BuildTarget) error {
-	workerCmd, _, testCmd, err := core.TestWorkerCommand(state, target)
+// verifyWorkerNotNeeded returns an error if a persistent worker is needed.
+func verifyWorkerNotNeeded(tid int, state *core.BuildState, target *core.BuildTarget) error {
+	workerCmd, _, _, err := core.TestWorkerCommand(state, target)
```
Comment (samwestmoreland) [inline src/test/test_step.go:550]: This eventually calls the sequence replacement stuff for `$(worker)` as well so maybe we can get rid of this?

**Review by samwestmoreland — APPROVED** [review]
Lgtm! I made a comment about the `$(worker)` code but now I'm thinking maybe better to leave that in for now so it's clearer to the user what's changed? Rather than just deleting it completely. That's probably what you were thinking?!

File: src/build/build_step.go (hunk 1 above)
Comment (goddenrich) [reply src/build/build_step.go:1218]: Yea my aim is to keep this in for now this allows us to give people hints to what is wrong. Without this code in core we aren't able to detect it so easily

**Pull request merged**